Octokit/Models/Request/Enterprise/UpdateMaintenanceRequest.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class UpdateMaintenanceRequest
+    {
+        /// <summary>
+        /// Maintenance request with default values (results in Maintenance mode being turned off immediately)
+        /// </summary>
+        public UpdateMaintenanceRequest()
+        { }
+
+        /// <summary>
+        /// Maintenance request specifying whether to enable/disable maintenance and when (only applicable when enabling maintenance)
+        /// </summary>
+        /// <param name="enabled">true to enable, false to disable</param>
+        /// <param name="when">A <see cref="MaintenanceDate"/> object specifying when maintenance mode will be enabled.
+        /// Use static helper methods
+        ///   <see cref="MaintenanceDate.Now()"/>
+        ///   <see cref="MaintenanceDate.FromDateTimeOffset(DateTimeOffset)"/> and
+        ///   <see cref="MaintenanceDate.FromChronicValue(string)"/></param>
+        /// <remarks>If enabled is false, the when parameter is ignored and maintenance is turned off immediately</remarks>
+        public UpdateMaintenanceRequest(bool enabled, MaintenanceDate when)
+        {
+            Enabled = enabled;
+            When = when;
+        }
+
+        public bool Enabled { get; protected set; }
+        
+        public MaintenanceDate When { get; protected set; }
+
+        [SuppressMessage("Microsoft.Globalization", "CA1304:SpecifyCultureInfo", MessageId = "System.String.ToLower")]
+        public string AsJsonString()
+        {
+            var json = string.Format(CultureInfo.InvariantCulture,
```

Comment:
It might sound excessive, but perhaps a `StringBuilder` in here would make some of this `Concat` code easier to read...